src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -169,6 +169,10 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <OutputType Condition=" '$(TargetType)' == 'Container' or '$(TargetType)' == 'DocumentContainer' ">library</OutputType>
     <OutputType Condition=" '$(OutputType)' == '' ">exe</OutputType>
 
+    <!-- Whether or not a reference assembly (or ONLY a reference assembly) is produced. -->
+    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == ''">false</ProduceReferenceAssembly>
+    <ProduceOnlyReferenceAssembly Condition="'$(ProduceOnlyReferenceAssembly)' == ''">false</ProduceOnlyReferenceAssembly>
```

Comment:
Is this change necessary?

Reply:
Well, the next change tests `ProduceOnlyReferenceAssembly` so defaulting it seemed reasonable. And it seemed equally reasonable to then have the defaulting of `ProduceReferenceAssembly` in the same place.

Reply:
This isn't _necessary_ but is also generally fine.

In MSBuild, there are no true booleans, only strings treated like booleans. As a result, each property that is logically a boolean can have three states:

1. `true`
2. `false`
3. Empty, unset, or set to something else.

if you control the code that reads a variable, you can use this to your advantage by acting as though it's got a "default":
|condition|effect
|----------------------------|-----------------------|
| `'$(Property)' == 'false'` | "Defaults" to `true`  |
| `'$(Property)' == 'true'`  | "Defaults" to `false` |

Since this isn't obvious, it's often a good idea to explicitly declare a default value, as is done here.

There's a downside, though, which is that explicitly declaring a value makes it harder for the user to have a default value of their own, defined with `Condition="'$(ProduceOnlyReferenceAssembly)' == ''"`. It's possible that users might have used that condition in `Directory.Build.targets` or NuGet `.targets`.

I don't see any of those in a [search for `ProduceOnlyReferenceAssembly extension:targets`](https://github.com/search?l=&p=2&q=ProduceOnlyReferenceAssembly+extension%3Atargets&ref=advsearch&type=Code), but there are too many hits to be totally confident.

As a result, and since this is targeting a release that is intended to minimize risk, **I think we should back out these changes and add only the `_DebugSymbolsProduced` check.**

Reply:
That's fair enough.

For boolean properties, is there a Microsoft convention and/or best practice for testing them? Always comparing to `'true'`? Always comparing to `'false'`? Always using equal tests for `'true'/'false'`? (The latter is probably something to avoid, because it would potentially match neither condition.)

And yes, it would be convenient to have some means of safely specifying property defaults in a .props file without also assigning at that point. Setting a bunch of properties in .targets always feels a bit naughty :)

Reply:
Done